sbi/inference/snle/snle_base.py

```diff
@@ -367,4 +367,4 @@ def _loss(self, theta: Tensor, x: Tensor) -> Tensor:
         Returns:
             Negative log prob.
         """
-        return -self._neural_net.log_prob(x, context=theta)
+        return self._neural_net.loss(x, condition=theta)
```

Comment:
I agree, `_density_estimator` would be clearer (maybe create an issue?)

Reply:
Depends on how we want to integrate the "non" density estimators. As this here is shared by any `NeuralInference` object, this would not be super clear for SNRE (ratios).